session/session.go

```diff
@@ -160,9 +161,35 @@ func New(ctx context.Context, fatalErrCh chan error, logBackend *log.Backend, cf
 	s.setTimers(doc)
 
 	s.Go(s.worker)
+	s.tq = NewTimerQ(DelayQueue{nextQ: s.egressQueue, s: s, Lambda: doc.SendLambda, Max: doc.SendMaxInterval})
 	return s, nil
 }
 
+type DelayQueue struct {
+	Queue
+	nextQ EgressQueue
+	s *Session
+	Lambda float64
+	Max uint64
+}
+
+func (d DelayQueue) Push(e *Message) error {
+	wakeMsec := uint64(rand.Exp(rand.NewMath(), d.Lambda))
+	switch {
```

Comment:
why use a switch statement here?

Reply:
this wasn't supposed to get merged either, this is just a quick hack to test chaining queues